include/multipass/passphrase_handler.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2021 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_PASSPHRASE_HANDLER_H
+#define MULTIPASS_PASSPHRASE_HANDLER_H
+
+#include "singleton.h"
+
+#include <QString>
+
+#define MP_PASSPHRASE_HANDLER multipass::PassphraseHandler::instance()
+
+namespace multipass
+{
+class PassphraseHandler : public Singleton<PassphraseHandler>
```

Comment:
Feels weird for this to be a singleton. Shouldn't it be part of `ClientConfig` instead, with an interface and a corresponding mock for testing?

Reply:
The `Daemon` uses it as well and so I thought it better to be it's own class and be a singleton since it's really unnecessary for the `Client` and `Daemon` classes to carry around an instance of this class for just a couple of calls.  And making it a singleton makes it easy to mock in tests.

Reply:
Should it go into the Utils singleton, then?

Reply:
If you prefer it not be explicit in what it does, then sure, that would be good place to put the `generate_hash_for()` function.  I'll probably rename that function to something like `generate_passphrase_hash_for()` in `Utils` in order to make it clearer what it is intended for,

Reply:
It isn't really bound to the passphrase, though, is it… It's just a "generate_scrypt_hash(data)"?

Reply:
Although I understand that by making it _about_ the passphrase makes for lower chance of mistakes and getting it out of sync when writing and reading… Still, I think `Utils` would be a better place since it's just a single method.

Reply:
Ok, moved it to `Utils`.